app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -46,6 +47,10 @@ def ems_inv_to_hashes
 
       private
 
+      def get_network_interfaces
+        @data[:network_interfaces] = gather_data_for_this_region(@nis)
```

Comment:
I know this is not related to this work, and GitHub does not allow me to directly comment on the `gather_data_for_this_region` method. But for that method we could use `collect` instead of `each`.

This refactor work looks good to me.